gosub-bindings/src/gosub_api.c

```diff
@@ -1,21 +1,16 @@
 #include "gosub_api.h"
 
-struct render_tree_t *render_tree_init(const char *html) {
-  struct render_tree_t *render_tree = malloc(sizeof(*render_tree));
-  if (!render_tree)
-    return NULL;
+void render_tree_init(struct render_tree_t *render_tree, const char *html) {
 
   render_tree->tree = gosub_render_tree_init(html);
   if (!render_tree->tree) {
-    free(render_tree);
-    return NULL;
+    return;
```

Comment:
Wouldn't it maybe make sense to return a integer to tell the caller if something went wrong?

Reply:
That's usually a common pattern in C, but I don't really think it'll be useful here... It doesn't matter to the caller whether the tree iterator or tree itself failed to allocate (at least, I don't know what exactly you would do with that level of detail)

Reply:
oh, I am too tired, I was still thinking we are returning a pointer, I forgot I changed this to void. Returning `-1` on failure and `0` on success is something I am good with in this case. I need some sleep :D

Reply:
When I first read that comment, I was thinking of an enum that described which part actually failed, not that a failure in general occurred. My bad

Reply:
All good, happens to me all the time... 😄